### DISCUSSION TITLE
Is there any volPath tracer implemented using the python API

### DISCUSSION THREAD

**Discussion by ioissss** [mitsuba-renderer/mitsuba3#1289] [Q&A] [answered]
Is there any volPath tracer implementation using python's API that supports volume rendering? Or a relevant reference? I would appreciate any replies!

**Comment by shinyoung-yi** [comment, marked as answer]
Hi, I'm a user of Mitsuba 3, but I'd be happy to help.

The PRBVolPath is a volumetric path tracer implemented in Python. You can find the implementation here:
[PRBVolPath code](https://github.com/mitsuba-renderer/mitsuba3/blob/master/src/python/python/ad/integrators/prbvolpath.py).

Some of the code in this file is dedicated to differentiable rendering (PRB). If you're looking for a simplified Python implementation that doesn't consider differentiability, you might be able to achieve this by removing the conditional branches that check for `mode != dr.ADMode.Primal`, where `mode` is a parameter of the `PRBVolpathIntegrator.sample()` method, to the best of my knowledge.

**Reply by ioissss** [reply]
Thank you very much for your answer, it helped me a lot!
